### PR TITLE
New Keyboard: Palette1202

### PR DESCRIPTION
**Pull request by niltea**
## Description

New PCB - Palette1202
* Left-hand devede for the illustrators who using [Photoshop](https://www.adobe.com/products/photoshop.html) or [Clip Studio Paint](https://www.clipstudio.net/en/).
* Includes Default keymap, and a keymap for operation check after assembly.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/palette1202/config.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0xFEED
+#define PRODUCT_ID   0x0000
```
Comment (drashna) [inline keyboards/palette1202/config.h]:
```suggestion
#define PRODUCT_ID   0x1202
```

File: keyboards/palette1202/config.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+/* Encoders */
+#define ENCODERS_PAD_A { F4, F6 }
+#define ENCODERS_PAD_B { F5, F7 }
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* Register custom font file */
+#ifdef OLED_DRIVER_ENABLE
+  #define OLED_FONT_H "./lib/glcdfont.c"
```
Comment (drashna) [inline keyboards/palette1202/config.h]: Technically, you don't need the `./` there, but it shouldn't make a difference.

```suggestion
  #define OLED_FONT_H "lib/glcdfont.c"
```

File: keyboards/palette1202/config.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+#endif
+
+/* Feature disable options */
+#define NO_DEBUG
+#define NO_PRINT
```
Comment (drashna) [inline keyboards/palette1202/config.h]: If you have console enabled, you may want to leave these on, actually.

File: keyboards/palette1202/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+    ),
+    [SETTING] = LAYOUT(
+      _______, KC_NO, Win_CS, Mac_CS, KC_NO,
+      _______, KC_NO, Win_PS, Mac_PS, KC_NO,
+               KC_NO, KC_NO,  KC_NO,  KC_NO
+    ),
+};
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+  int currentDefault = eeconfig_read_default_layer();
```
Comment (drashna) [inline keyboards/palette1202/keymaps/default/keymap.c]:
```suggestion
  uint8_t currentDefault = get_highest_layer(defaultlayer_state);
```

File: keyboards/palette1202/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+  int currentLayer = get_highest_layer(layer_state);
```
Comment (drashna) [inline keyboards/palette1202/keymaps/default/keymap.c]:
```suggestion
  uint8_t currentLayer = get_highest_layer(layer_state);
```

File: keyboards/palette1202/readme.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Palette1202
+
+---
+
+![Palette1202](https://palette1202.nilgiri-tea.net/images/palette1202.jpg)
+
+A left hand device with rotary encoder, for artists.
+
+Keyboard Maintainer: [niltea](https://github.com/niltea)
```
Comment (drashna) [inline keyboards/palette1202/readme.md]:
```suggestion
* Keyboard Maintainer: [niltea](https://github.com/niltea)  
```

File: keyboards/palette1202/readme.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Palette1202
+
+---
+
+![Palette1202](https://palette1202.nilgiri-tea.net/images/palette1202.jpg)
+
+A left hand device with rotary encoder, for artists.
+
+Keyboard Maintainer: [niltea](https://github.com/niltea)  
+Hardware Supported: Palette1202
```
Comment (drashna) [inline keyboards/palette1202/readme.md]:
```suggestion
* Hardware Supported: Palette1202
```

File: keyboards/palette1202/readme.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+Hardware Availability: [Pixiv Booth](https://booth.pm/)
```
Comment (drashna) [inline keyboards/palette1202/readme.md]:
```suggestion
* Hardware Availability: [Pixiv Booth](https://booth.pm/)
```

File: keyboards/palette1202/readme.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+[PCB & Case](https://github.com/niltea/Palette1202)
+
+Make example for this keyboard (after setting up your build environment):
+
+- default
+
+		make palette1202:default:avrdude
```
Comment (drashna) [inline keyboards/palette1202/readme.md]: We have a universal flash target now. No need to specify a specific target

```suggestion
		make palette1202:default:flash
```

File: keyboards/palette1202/readme.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+        
+- key check (for after build)
+
+		make palette1202:key-check:avrdude
```
Comment (drashna) [inline keyboards/palette1202/readme.md]:
```suggestion
		make palette1202:key-check:flash
```

File: keyboards/palette1202/rules.mk [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+MIDI_ENABLE        = no     # MIDI support
+BLUETOOTH_ENABLE   = no     # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE       = no     # Audio output on port C6
+FAUXCLICKY_ENABLE  = no     # Use buzzer to emulate clicky switches
+HD44780_ENABLE     = no     # Enable support for HD44780 based LCDs
+ENCODER_ENABLE     = yes    # Enable support for rotary encoders
+OLED_DRIVER_ENABLE = yes    # Enable support for OLED display
+
+# Additional code
+SRC += ./lib/oled_helper.c  # Adding OLED
```
Comment (drashna) [inline keyboards/palette1202/rules.mk]:
```suggestion
SRC += lib/oled_helper.c  # Adding OLED
```

File: keyboards/palette1202/keymaps/default/keymap.c (hunk 4 above)
Comment (noroadsleft) [reply keyboards/palette1202/keymaps/default/keymap.c]: Shouldn't it be `default_layer_state` with two underscores?

File: keyboards/palette1202/info.json [hunk 12]
```diff
@@ -0,0 +1,30 @@
+{
+  "keyboard_name": "Palette1202",
+  "url": "https://github.com/niltea/Palette1202",
+  "maintainer": "niltea",
+  "width": 5,
+  "height": 3.5,
+  "layouts": {
+    "LAYOUT": {
+      "key_count": 12,
```
Comment (noroadsleft) [inline keyboards/palette1202/info.json]:
```suggestion
      "key_count": 14,
```

File: keyboards/palette1202/readme.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
+# Palette1202
+
+---
+
```
Comment (noroadsleft) [inline keyboards/palette1202/readme.md]: You don't really *need* this horizontal rule here; GitHub adds a thin bottom border to the header you have on Line 1, so you could delete Lines 3-4 if you wish.

File: keyboards/palette1202/rules.mk [hunk 14]
```diff
@@ -0,0 +1,25 @@
+MCU = atmega32u4
```
Comment (noroadsleft) [inline keyboards/palette1202/rules.mk]:
```suggestion
# MCU name
MCU = atmega32u4
```

File: keyboards/palette1202/rules.mk [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+BOOTLOADER = atmel-dfu
```
Comment (noroadsleft) [inline keyboards/palette1202/rules.mk]:
```suggestion
# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
BOOTLOADER = atmel-dfu
```

File: keyboards/palette1202/rules.mk [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+
+# Build Options
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (noroadsleft) [inline keyboards/palette1202/rules.mk]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```

File: keyboards/palette1202/keymaps/default/keymap.c (hunk 4 above)
Comment (drashna) [reply keyboards/palette1202/keymaps/default/keymap.c]: yes

File: keyboards/palette1202/config.h (hunk 3 above)
Comment (niltea) [reply keyboards/palette1202/config.h]: It is ok, thanks! (or should it be enabled?)

File: keyboards/palette1202/readme.md (hunk 13 above)
Comment (niltea) [reply keyboards/palette1202/readme.md]: I'll remove this, thanks!

File: keyboards/palette1202/readme.md (hunk 9 above)
Comment (niltea) [reply keyboards/palette1202/readme.md]: I didn't know it, thanks!

File: keyboards/palette1202/lib/glcdfont.c [hunk 17]
```diff
@@ -0,0 +1,244 @@
...
+#define FONT5X7_H
+
+#ifdef __AVR__
+ #include <avr/io.h>
+ #include <avr/pgmspace.h>
+#elif defined(ESP8266)
+ #include <pgmspace.h>
+#else
+ #define PROGMEM
+#endif
```
Comment (fauxpark) [inline keyboards/palette1202/lib/glcdfont.c]: Please replace this block with:
```c
#include "progmem.h"
```
per #7745

File: keyboards/palette1202/lib/glcdfont.c [hunk 18]
```diff
@@ -0,0 +1,244 @@
...
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+#endif // FONT5X7_H
```
Comment (fauxpark) [inline keyboards/palette1202/lib/glcdfont.c]:
```suggestion
```

File: keyboards/palette1202/lib/oled_helper.h [hunk 19]
```diff
@@ -0,0 +1,9 @@
+#ifdef OLED_DRIVER_ENABLE
+#ifndef OLED_HELPER_H
```
Comment (fauxpark) [inline keyboards/palette1202/lib/oled_helper.h]:
```suggestion
```

File: keyboards/palette1202/lib/oled_helper.h [hunk 20]
```diff
@@ -0,0 +1,9 @@
...
+#define OLED_HELPER_H
+
+void render_row(int row, const char* status);
+
+#endif /* #ifndef OLED_HELPER_H */
```
Comment (fauxpark) [inline keyboards/palette1202/lib/oled_helper.h]:
```suggestion
```

File: keyboards/palette1202/lib/oled_helper.h [hunk 21]
```diff
@@ -0,0 +1,9 @@
+#ifdef OLED_DRIVER_ENABLE
+#ifndef OLED_HELPER_H
+#define OLED_HELPER_H
```
Comment (fauxpark) [inline keyboards/palette1202/lib/oled_helper.h]:
```suggestion
```

File: keyboards/palette1202/lib/oled_helper.h [hunk 22]
```diff
@@ -0,0 +1,9 @@
+#ifdef OLED_DRIVER_ENABLE
```
Comment (fauxpark) [inline keyboards/palette1202/lib/oled_helper.h]:
```suggestion
#pragma once

#ifdef OLED_DRIVER_ENABLE
```

File: keyboards/palette1202/readme.md [hunk 23]
```diff
@@ -0,0 +1,22 @@
+# Palette1202
+
+![Palette1202](https://palette1202.nilgiri-tea.net/images/palette1202.jpg)
+
+A left hand device with rotary encoder, for artists.
+
+* Keyboard Maintainer: [niltea](https://github.com/niltea)
```
Comment (fauxpark) [inline keyboards/palette1202/readme.md]:
```suggestion
* Keyboard Maintainer: [niltea](https://github.com/niltea)
```

File: keyboards/palette1202/readme.md [hunk 24]
```diff
@@ -0,0 +1,22 @@
...
+* Keyboard Maintainer: [niltea](https://github.com/niltea)  
+* Hardware Supported: Palette1202
+* Hardware Availability: [Pixiv Booth](https://booth.pm/)
+* [PCB & Case](https://github.com/niltea/Palette1202)
+
+Make example for this keyboard (after setting up your build environment):
+
+- default
+
+		make palette1202:default:flash
```
Comment (fauxpark) [inline keyboards/palette1202/readme.md]:
```suggestion
    make palette1202:default:flash
```

File: keyboards/palette1202/readme.md [hunk 25]
```diff
@@ -0,0 +1,22 @@
...
+        
+- key check (for after build)
+
+		make palette1202:key-check:flash
```
Comment (fauxpark) [inline keyboards/palette1202/readme.md]:
```suggestion
    make palette1202:key-check:flash
```

File: keyboards/palette1202/lib/glcdfont.c (hunk 17 above)
Comment (niltea) [reply keyboards/palette1202/lib/glcdfont.c]: okay, thanks.

File: keyboards/palette1202/config.h (hunk 3 above)
Comment (drashna) [reply keyboards/palette1202/config.h]: Basically, these defines should be removed.  

If console isn't enabled, they're defined anyways (and may cause compilation issues).

File: keyboards/palette1202/palette1202.c [hunk 26]
```diff
@@ -0,0 +1,30 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "palette1202.h"
+
+// set base layer and save it to EPROM
+void setBaseLayer(uint16_t layerID) {
+  eeconfig_update_default_layer(1UL<<layerID);
+  default_layer_set(1UL<<layerID);
+}
```
Comment (drashna) [inline keyboards/palette1202/palette1202.c]: This functionality already exists, actually: https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/palette1202/palette1202.h [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+    k10, k11, k12, k13, k14,       \
+         k21, k22, k23, k24        \
+)                                  \
+{                                  \
+    { k00,   k01, k02, k03, k04 }, \
+    { k10,   k11, k12, k13, k14 }, \
+    { KC_NO, k21, k22, k23, k24 }, \
+}
+
+void setBaseLayer(uint16_t layerID);
```
Comment (drashna) [inline keyboards/palette1202/palette1202.h]:
```suggestion
```

File: keyboards/palette1202/keymaps/default/keymap.c [hunk 28]
```diff
@@ -0,0 +1,282 @@
...
+  }
+}
+
+// custom keycode
+// switch default layer
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case Mac_CS:
+      if (record->event.pressed) {
+        setBaseLayer(MAC_CS_1);
```
Comment (drashna) [inline keyboards/palette1202/keymaps/default/keymap.c]:
```suggestion
        set_single_persistent_default_layer(MAC_CS_1);
```

File: keyboards/palette1202/keymaps/default/keymap.c [hunk 29]
```diff
@@ -0,0 +1,282 @@
...
+      }
+      return false;
+      break;
+    case Mac_PS:
+      if (record->event.pressed) {
+        setBaseLayer(MAC_PS_1);
```
Comment (drashna) [inline keyboards/palette1202/keymaps/default/keymap.c]:
```suggestion
        set_single_persistent_default_layer(MAC_PS_1);
```

File: keyboards/palette1202/keymaps/default/keymap.c [hunk 30]
```diff
@@ -0,0 +1,282 @@
...
+      }
+      return false;
+      break;
+    case Win_CS:
+      if (record->event.pressed) {
+        setBaseLayer(WIN_CS_1);
```
Comment (drashna) [inline keyboards/palette1202/keymaps/default/keymap.c]:
```suggestion
        set_single_persistent_default_layer(WIN_CS_1);
```

File: keyboards/palette1202/keymaps/default/keymap.c [hunk 31]
```diff
@@ -0,0 +1,282 @@
...
+      }
+      return false;
+      break;
+    case Win_PS:
+      if (record->event.pressed) {
+        setBaseLayer(WIN_PS_1);
```
Comment (drashna) [inline keyboards/palette1202/keymaps/default/keymap.c]:
```suggestion
        set_single_persistent_default_layer(WIN_PS_1);
```

File: keyboards/palette1202/rules.mk [hunk 32]
```diff
@@ -0,0 +1,36 @@
+# MCU name
+MCU = atmega32u4
+# Bootloader selection
```
Comment (fauxpark) [inline keyboards/palette1202/rules.mk]:
```suggestion

# Bootloader selection
```

File: keyboards/palette1202/lib/oled_helper.h [hunk 33]
```diff
@@ -0,0 +1,7 @@
...
+#pragma once
```
Comment (noroadsleft) [inline keyboards/palette1202/lib/oled_helper.h]: The `#pragma once` statement should go first.

```suggestion
#pragma once
#ifdef OLED_DRIVER_ENABLE
```

**Pull request merged**